src/Simplex/Messaging/Agent.hs

```diff
@@ -271,9 +260,54 @@ processCommand c@AgentClient {sndQ} st (ATransmission corrId entity cmd) =
             agentMessage = REPLY qInfo
           }
 
-    respond :: EntityCommand t c => Entity t -> ACommand 'Agent c -> m ()
+    respond :: EntityCommand t c' => Entity t -> ACommand 'Agent c' -> m ()
     respond ent resp = atomically . writeTBQueue sndQ $ ATransmission corrId ent resp
 
+sendMessage :: forall m. AgentMonad m => AgentClient -> SQLiteStore -> ACorrId -> Entity 'Conn_ -> MsgBody -> m ()
+sendMessage c st corrId (Conn cId) msgBody =
+  withStore (getConn st cId) >>= \case
+    SomeConn _ (DuplexConnection _ _ sq) -> sendMsg sq
+    SomeConn _ (SndConnection _ sq) -> sendMsg sq
+    _ -> throwError $ CONN SIMPLEX
+  where
+    sendMsg :: SndQueue -> m ()
+    sendMsg sq = do
+      internalTs <- liftIO getCurrentTime
+      (internalId, internalSndId, previousMsgHash) <- withStore $ updateSndIds st sq
+      let msgStr =
+            serializeSMPMessage
+              SMPMessage
+                { senderMsgId = unSndId internalSndId,
+                  senderTimestamp = internalTs,
+                  previousMsgHash,
+                  agentMessage = A_MSG msgBody
+                }
+          msgHash = C.sha256Hash msgStr
+      withStore $
+        createSndMsg st sq $
+          SndMsgData {internalId, internalSndId, internalTs, msgBody, internalHash = msgHash}
+      sendAgentMessage c sq msgStr
+      atomically . writeTBQueue (sndQ c) $ ATransmission corrId (Conn cId) $ SENT (unId internalId)
+
+processBroadcastCommand ::
+  forall c m. (AgentMonad m, EntityCommand 'Broadcast_ c) => AgentClient -> SQLiteStore -> ACorrId -> Entity 'Broadcast_ -> ACommand 'Client c -> m ()
+processBroadcastCommand c st corrId bcast@(Broadcast bId) = \case
+  NEW -> withStore (createBcast st bId) >> ok
+  ADD (Conn cId) -> withStore (addBcastConn st bId cId) >> ok
+  REM (Conn cId) -> withStore (removeBcastConn st bId cId) >> ok
```

Comment:
it feels like using entity for connection here is wrong (`Conn`), since we refer to entity as smth that prepends command and to which the command is applied (bcast), and this is only a parameter for command. How about switching these to using `ConnAlias`? - It is even expressed then as `ConnAlias` in store

Reply:
I think it's better like that as it allows for future extension and also clarifies what it is.